typescript/polyglotpkg/src/lib/model.ts

```diff
@@ -6,116 +6,156 @@
  * %%
  * Build Tools for VMware Aria
  * Copyright 2023 VMware, Inc.
- * 
- * This product is licensed to you under the BSD-2 license (the "License"). You may not use this product except in compliance with the BSD-2 License.  
- * 
+ *
+ * This product is licensed to you under the BSD-2 license (the "License"). You may not use this product except in compliance with the BSD-2 License.
+ *
  * This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
  * #L%
  */
 import { Writable } from "stream";
 
 export enum ActionType {
-    UNKNOWN = 'unknown',
-    VRO = 'vro',
-    ABX = 'abx',
+	UNKNOWN = 'unknown',
+	VRO = 'vro',
+	ABX = 'abx',
 }
 
+// This section contains a list of supported runtimes for actions. Adding a new runtime requires updating this section and this section only.
+// If this is ever not the case, we need to refactor the code to make it so.
+
 export enum ActionRuntime {
-    UNKNOWN = 'unknown',
-    VRO_NODEJS_12 = 'node:12',
-    VRO_NODEJS_14 = 'node:14',
-    VRO_POWERCLI_11_PS_62 = 'powercli:11-powershell-6.2',
-    VRO_POWERCLI_12_PS_71 = 'powercli:12-powershell-7.1',
-    VRO_PYTHON_37 = 'python:3.7',
-    VRO_PYTHON_310 = 'python:3.10',
-    ABX_NODEJS = 'nodejs',
-    ABX_POWERSHELL = 'powershell',
-    ABX_PYTHON = 'python',
+	UNKNOWN = 'unknown',
+	VRO_NODEJS_12 = 'node:12',
+	VRO_NODEJS_14 = 'node:14',
+	VRO_NODEJS_18 = 'node:18',
+	VRO_NODEJS_20 = 'node:20',
+
+	VRO_POWERCLI_11_PS_62 = 'powercli:11-powershell-6.2',
+	VRO_POWERCLI_12_PS_71 = 'powercli:12-powershell-7.1',
+
+	VRO_PYTHON_37 = 'python:3.7',
+	VRO_PYTHON_310 = 'python:3.10',
+
+	ABX_NODEJS = 'nodejs',
+	ABX_POWERSHELL = 'powershell',
+	ABX_PYTHON = 'python',
 }
 
-export type ActionRuntimeType =
-    ActionRuntime.ABX_NODEJS | ActionRuntime.VRO_NODEJS_12 | ActionRuntime.VRO_NODEJS_14
-    | ActionRuntime.ABX_POWERSHELL | ActionRuntime.VRO_POWERCLI_11_PS_62 | ActionRuntime.VRO_POWERCLI_12_PS_71
-    | ActionRuntime.VRO_PYTHON_37 | ActionRuntime.VRO_PYTHON_310 | ActionRuntime.ABX_PYTHON
+// After adding a new runtime, add it to the appropriate list below.
+
+export const VroNodeJsActionRuntimes = [
+	ActionRuntime.VRO_NODEJS_12,
+	ActionRuntime.VRO_NODEJS_14,
+	ActionRuntime.VRO_NODEJS_18,
+	ActionRuntime.VRO_NODEJS_20
+];
+
+export const VroPowershellActionRuntimes = [
+	ActionRuntime.VRO_POWERCLI_11_PS_62,
+	ActionRuntime.VRO_POWERCLI_12_PS_71
+];
+
+export const VroPythonActionRuntimes = [
+	ActionRuntime.VRO_PYTHON_37,
+	ActionRuntime.VRO_PYTHON_310
+];
+
+export const NodeJsActionRuntimes = VroNodeJsActionRuntimes.concat(...[ActionRuntime.ABX_NODEJS]);
```

Comment:
Why do we need both spread and concat? Don't we need either
```` VroNodeJsActionRuntimes.concat(ActionRuntime.ABX_NODEJS) ```
or
``` [...VroNodeJsActionRuntimes, ActionRuntime.ABX_NODEJS]```

I might be wrong though...

Reply:
Great catch, I think I was mind flooding myself :D